docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -226,6 +226,30 @@ AnnotationsTransformerBuildItem transform() {
 }
 ----
 
+=== Additional Interceptor Bindings
+
+In rare cases it might be handy to programmatically register an existing annotation as interceptor binding.
```

Comment:
Hey @manovotn ,
Do you mind adding a concrete example of when one would want to do this. We want the Quarkus docs to be concrete an example driven vs abstract like a reference guide might be.
I tried to find one but was not sure of a use case for it.

Reply:
It might be a bit corner case-ish to add that here, that's why I just added the notion of similarity to CDI's `BeforeBeanDiscovery#addInterceptorBinding()`.

You would use this when you need to change an annotation you have no control over - one from 3rd party library for instance. This way you can turn it into an interceptor binding and write an interceptor that will handle all calls going to methods with that annotation.

Either way, I am not the author of the original issue, @mkouba is, so I suppose some integration we intended to do needed this?

Reply:
Looks like the original use case was SR FT integration but that was done differently in the end.